actix-http/src/extensions.rs

```diff
@@ -70,6 +70,92 @@ impl fmt::Debug for Extensions {
     }
 }
 
+#[test]
+fn test_remove() {
+    let mut map = Extensions::new();
+
+    map.insert::<i8>(123);
+    assert!(map.get::<i8>().is_some());
+
+    map.remove::<i8>();
+    assert!(!map.get::<i8>().is_some());
+}
+
+#[test]
+fn test_clear() {
+    let mut map = Extensions::new();
+
+    map.insert::<i8>(8);
+    map.insert::<i16>(16);
+    map.insert::<i32>(32);
+
+    assert!(map.contains::<i8>());
+    assert!(map.contains::<i16>());
+    assert!(map.contains::<i32>());
+
+    map.clear();
+
+    assert!(!map.contains::<i8>());
+    assert!(!map.contains::<i16>());
+    assert!(!map.contains::<i32>());
+
+    map.insert::<i8>(10);
+    assert_eq!(*map.get::<i8>().unwrap(), 10);
+}
+
+#[test]
+fn test_integers() {
+    let mut map = Extensions::new();
+
+    map.insert::<i8>(8);
+    map.insert::<i16>(16);
+    map.insert::<i32>(32);
+    map.insert::<i64>(64);
+    map.insert::<i128>(128);
+    map.insert::<u8>(8);
+    map.insert::<u16>(16);
+    map.insert::<u32>(32);
+    map.insert::<u64>(64);
+    map.insert::<u128>(128);
+    assert!(map.get::<i8>().is_some());
+    assert!(map.get::<i16>().is_some());
+    assert!(map.get::<i32>().is_some());
+    assert!(map.get::<i64>().is_some());
+    assert!(map.get::<i128>().is_some());
+    assert!(map.get::<i8>().is_some());
+    assert!(map.get::<i16>().is_some());
+    assert!(map.get::<i32>().is_some());
+    assert!(map.get::<i64>().is_some());
+    assert!(map.get::<i128>().is_some());
+}
+
+#[test]
+fn test_composition() {
+    struct Magi<T>(pub T);
+
+    struct Madoka {
+        pub god: bool,
+    }
+
+    struct Homura {
+        pub attempts: usize,
+    }
+
+    struct Mami {
+        pub guns: usize,
+    }
+
+    let mut map = Extensions::new();
+
+    map.insert(Magi(Madoka { god: false }));
+    map.insert(Magi(Homura { attempts: 0 }));
+    map.insert(Magi(Mami { guns: 999 }));
+
+    assert_eq!(false, map.get::<Magi<Madoka>>().unwrap().0.god);
```

Comment:
Yeah, the current assertion isn't wrong but I'd prefer using `assert!` to `assert_eq!` in this case.

```suggestion
    assert!(!map.get::<Magi<Madoka>>().unwrap().0.god);
```